packages/components/src/popover/index.js

```diff
@@ -273,14 +276,40 @@ const Popover = ( {
 				contentRect.current = contentRef.current.getBoundingClientRect();
 			}
 
+			const { offsetParent, ownerDocument } = containerRef.current;
+			let relativeOffsetTop = 0;
+
+			// If there is a positioned ancestor element that is not the body,
+			// subtract the position from the anchor rect.
+			// See https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetParent
+			if ( offsetParent && offsetParent !== ownerDocument.body ) {
+				const offsetParentRect = offsetParent.getBoundingClientRect();
+
+				relativeOffsetTop = offsetParentRect.top;
+				anchor = new window.DOMRect(
+					anchor.left - offsetParentRect.left,
+					anchor.top - offsetParentRect.top,
+					anchor.width,
+					anchor.height
+				);
+			}
+
 			const {
 				popoverTop,
 				popoverLeft,
 				xAxis,
 				yAxis,
 				contentHeight,
 				contentWidth,
-			} = computePopoverPosition( anchor, contentRect.current, position, __unstableSticky, anchorRef );
+			} = computePopoverPosition( anchor, contentRect.current, position, __unstableSticky, anchorRef, relativeOffsetTop );
+
+			let cortainerPosition;
+
+			if ( ! __unstableFixedPosition ) {
+				cortainerPosition = 'absolute';
+			}
+
+			setStyle( containerRef.current, 'position', cortainerPosition );
```

Comment:
Not within the scope of this pull request, but I'm not sure I understand the purpose of `setStyle`.

- It claims to be used for "removing" a style, but it's logic only appears to contain direct assignment
- Is there some reason we're wanting to avoid to assign directly into the style, even if the value is the same? Is there a performance implication?

Reply:
It's used for setting (and removing) a style. We need to set it directly because React updates the styles async, so there's a noticeable delay when you scroll for example.

Reply:
But it's not clear to me what value it brings over, say...

```js
containerRef.current.style.position = containerPosition;
```

Reply:
Not continuously setting the style? I guess browsers do have built in optimisations when setting the same style.